<?xml version="1.0"?>

<launch>
  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

  <!-- Run a passthrough filter to clean NaNs -->
  <node pkg="nodelet" type="nodelet" name="passthroughZ" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/sick_3vistort_driver/points" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: 0.4
      filter_limit_max: 3.0
      filter_limit_negative: False
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="passthroughY" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/passthroughZ/output" />
    <rosparam>
      filter_field_name: y
      filter_limit_min: -0.05
      filter_limit_max: 3.0
      filter_limit_negative: False
    </rosparam>
  </node>

  <node pkg="nodelet" type="nodelet" name="statisticaloutlierremoval" args="load pcl/StatisticalOutlierRemoval pcl_manager" output="screen">
    <remap from="~input" to="/passthroughY/output" />
    <rosparam>
      mean_k: 5
      stddev: 0.5
    </rosparam>
    <remap from="statisticaloutlierremoval/output" to="/sick_3vistort_driver/filtered_points"/>
  </node>

</launch>